cpp/lib/common/error_message.cpp

```diff
@@ -32,7 +32,7 @@ namespace TypeDB {
 
 namespace DriverError {
 
-#define ERR_DRIVER(ID, MSG) ERRMSG("CCL", "Driver Error", ID, MSG)
```

Comment:
Git went crazy, so I duplicate the conversation here.

The latest comment from @dmitrii-ubskii:

> The reason we introduced language prefixes in codes is so that we don't run into collisions between the Rust and the host language's error codes, mainly for internal errors. INT02 coming from Java's ""Illegal state has been reached!" is distinct from Rust INT02 "Unable to send response over callback channel (receiver dropped)."
We then reasoned that there's no guarantee that Rust driver won't be emitting its own Concept errors, e.g., so the prefix got added to all Java error codes. That should have also been the case in Python. Node is the only place where collisions are irrelevant as it's not using FFI.

My reply:
I had a talk about it with @flyingsilverfin. We thought that we are okay with the collisions of the error codes as far as they are just ideological, not real collision problems within the code (each driver runs separately). And for each error code we have context of the driver if somebody wants to check what to do in each specific situation.

I'm okay with two ways:
* Having language-based prefix for **all** error codes we have in drivers
* Not having one. 

I don't really like exceptions like Nodejs, because the need of differentiating FFI/non-FFI drivers only complicates things here and a user receiving such errors is not interested if a driver they are using is based on FFI or not.

Are there technical problems with having same error codes with different content for different drivers?

Reply:
* We have checks in cloud tests that compare error codes.
* UX and error reporting suffers when users search for the error code and find something entirely different.

Reply:
* We do not have checks that compare error codes, they would fail on factory otherwise. Only steps with "exception contains substring"
* If we care about UX, then we still have to change python and nodejs codes as well (they were outdated).

@flyingsilverfin has @dmitrii-ubskii been convincing enough to accept change to "CCDR" (cpp), "CSDR" (csharp), "JDR" (java), "PDR" (python), "NDR" (node?..) (or "TDR" as typescript?..)?

Reply:
I mean yeah i don't mind using a unique prefix for each! We can do that :)
C++ - CCDR
C# - CSDR
Java - JDR
Rust - RDR
Node - NDR
Python - PDR

If C has its own then we can use CDR, but i think it doesn't?

Reply:
Yeah, it does not. Okay, I proceed with this plan then, thanks!

Reply:
> * We do not have checks that compare error codes, they would fail on factory otherwise. Only steps with "exception contains substring"

Sorry, I meant in the typedb-cloud repo.